dali/c_api/c_api.cc

```diff
@@ -82,21 +87,35 @@ void SetExternalInputTensors(daliPipelineHandle *pipe_handle, const char *name,
 
 }  // namespace
 
+daliPipelineHandle *daliInitialize() {
+  if (dali_initialized) {
+    return nullptr;
+  }
+  dali::InitOperatorsLib();
+  dali::DALIInit(dali::OpSpec("CPUAllocator"),
+                 dali::OpSpec("PinnedCPUAllocator"),
+                 dali::OpSpec("GPUAllocator"));
+  dali_initialized = true;
+  return new daliPipelineHandle;
```

Comment:
Why allocate here? There is no deallocate, this is a leak.

Reply:
Dealocation is in `daliDeletePipeline`. Typical approach in C APIs ;)

Reply:
This function is quite broken on many fronts:
1. Library initialization should be separate from pipeline creation.
2. The object returned is, well, a handle - I think that dynamically allocating _a handle_ makes very little sense, since a handle is by definition a lightweight object.
3. There's no reverse operation, so it  leaks resources.
and last but not least
4. It breaks daliDeletePipeline which assumed caller's ownership of the handle object.

If we were to follow typical C API, then we'd have:
in the header
```c++
typedef struct DALIPipeline *daliPipelineHandle;
```
in implementation
```c++
struct DALIPipeline {
  Pipeline *pipe;
  Workspace *ws;
};
```
daliPipelineHandle, being a pointer, would be passed by value everywhere except to daliCreatePipeline, where we'd allocate this small struct and delete it in daliDeletePipeline.

Reply:
Fixed